src/main/java/me/i509/fabric/bulkyshulkies/extension/QuickShulkerHook.java

```diff
@@ -0,0 +1,81 @@
+package me.i509.fabric.bulkyshulkies.extension;
+
+import me.i509.fabric.bulkyshulkies.BulkyShulkies;
+import me.i509.fabric.bulkyshulkies.api.block.base.AbstractShulkerBoxBlock;
+import me.i509.fabric.bulkyshulkies.client.screen.Generic11x7Screen;
+import me.i509.fabric.bulkyshulkies.client.screen.Generic13x7Screen;
+import me.i509.fabric.bulkyshulkies.client.screen.Generic9x7Screen;
+import me.i509.fabric.bulkyshulkies.client.screen.ScrollableScreen;
+import me.i509.fabric.bulkyshulkies.container.*;
+import me.i509.fabric.bulkyshulkies.registry.ShulkerBlocks;
+import net.fabricmc.fabric.api.client.screen.ScreenProviderRegistry;
+import net.fabricmc.fabric.api.container.ContainerProviderRegistry;
+import net.kyrptonaught.quickshulker.api.ItemStackInventory;
+import net.kyrptonaught.quickshulker.api.QuickOpenableRegistry;
+import net.kyrptonaught.quickshulker.api.RegisterQuickShulker;
+import net.minecraft.container.ShulkerBoxSlot;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.BlockItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.PacketByteBuf;
+
+public class QuickShulkerHook implements RegisterQuickShulker {
+   public static final Identifier QS$SHULKER_SCROLLABLE_CONTAINER = BulkyShulkies.id("qs_shulker_scrollable_container");
+   public static final Identifier QS$SHULKER_9x7_CONTAINER = BulkyShulkies.id("qs_shulker_container_9x7");
+   public static final Identifier QS$SHULKER_11x7_CONTAINER = BulkyShulkies.id("qs_shulker_container_11x7");
+   public static final Identifier QS$SHULKER_13x7_CONTAINER = BulkyShulkies.id("qs_shulker_container_13x7");
+   @Override
+   public void registerProviders() {
+      ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_SCROLLABLE_CONTAINER, QuickShulkerHook::createScrollable);
+      ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_9x7_CONTAINER, QuickShulkerHook::create9x7);
+      ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_11x7_CONTAINER, QuickShulkerHook::create11x7);
+      ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_13x7_CONTAINER, QuickShulkerHook::create13x7);
+
+      ScreenProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_9x7_CONTAINER, Generic9x7Screen::createScreen);
+      ScreenProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_11x7_CONTAINER, Generic11x7Screen::createScreen);
+      ScreenProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_13x7_CONTAINER, Generic13x7Screen::createScreen);
+      ScreenProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_SCROLLABLE_CONTAINER, ScrollableScreen::createScreen);
```

Comment:
I want this to work fine on servers, so possibly make a Client entrypoint. Also my screens have an Environment enchantment so this will fail.

Reply:
Also make sure this class uses tabs

Reply:
This will be addressed before release, not sure how I would like to approach this. Looking into alternatives

Reply:
One option is a Mixin Config plugin and a separate mixin which is only loaded on a client and when QuickShulker is present. If you do this please make sure you inject into your own QuickShulkerHook with `remap=false`

I have a trinkets example here (https://github.com/i509VCB/Project-F/blob/master/src/main/java/me/i509/fabric/projectf/mixin/compat/CompatMixinPlugin.java), but use a call to fabric loader first to verify we have a client and your mod is present.